src/shared/Calendar/Calendar.tsx

```diff
@@ -0,0 +1,228 @@
+import { green, red, orange } from "@mui/material/colors";
+import { UserData } from "../../context/UserDataContext";
+import { IconButton, Box } from "@mui/material";
+import ArrowBackIosIcon from "@mui/icons-material/ArrowBackIos";
+import ArrowForwardIosIcon from "@mui/icons-material/ArrowForwardIos";
+import dayjs, { Dayjs } from "dayjs";
+
+const Frame = styled("div")({
+  width: "900px",
+  border: "1px solid lightgrey",
+  boxShadow: "2px 2px 2px #eee",
+  margin: "0 auto",
+});
+
+const Header = styled("div")(({ theme }) => ({
+  fontSize: "17px",
+  fontWeight: "bold",
+  padding: "10px 10px 5px 10px",
+  display: "flex",
+  justifyContent: "space-around",
+  alignItems: "center",
+  backgroundColor: theme.palette.primary.main,
+  color: "white",
+}));
+
+const Body = styled("div")({
+  width: "100%",
+  padding: "20px",
+  display: "flex",
+  flexWrap: "wrap",
+});
+
+const WeekDays = styled("div")(({ theme }) => ({
+  width: "14%",
+  height: "40px",
+  display: "flex",
+  alignItems: "center",
+  justifyContent: "center",
+  cursor: "pointer",
+  margin: "1px",
+  color: theme.palette.primary.main,
+}));
+const Day = styled("button")(({ theme }) => ({
+  width: "14%",
+  height: "40px",
+  display: "flex",
+  alignItems: "center",
+  justifyContent: "center",
+  cursor: "pointer",
+  margin: "1px",
+  color: theme.palette.primary.main,
+}));
+
+const HeaderBox = styled("div")(({ theme }) => ({
+  width: "900px",
+  margin: "0 auto",
+  height: "40px",
+  display: "flex",
+  alignItems: "center",
+}));
+
+const ColorPoint = styled("div")(({ theme }) => ({
+  width: "12px",
+  height: "12px",
+  borderRadius: "50px",
+  margin: "0px 7px 0px 30px",
+}));
+
+const Calendar = () => {
+  const DAYS = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+  const DAYS_LEAP = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+  const DAYS_OF_THE_WEEK = ["MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN"];
+  const MONTHS = [
+    "JAN",
+    "FEB",
+    "MAR",
+    "APR",
+    "MAY",
+    "JUN",
+    "JUL",
+    "AUG",
+    "SEP",
+    "OCT",
+    "NOV",
+    "DEC",
+  ];
+
+  const today = dayjs();
+  const [date, setDate] = useState(today);
+  const [day, setDay] = useState(date.get("date"));
+  const [month, setMonth] = useState(date.get("month"));
+  const [year, setYear] = useState(date.get("year"));
+  const [startDay, setStartDay] = useState(getStartDayOfMonth(date));
+  const { data } = useContext(UserData);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    setDay(date.get("date"));
+    setMonth(date.get("month"));
+    setYear(date.get("year"));
+    setStartDay(getStartDayOfMonth(date));
+  }, [date]);
+
+  function getStartDayOfMonth(date: Dayjs) {
+    return dayjs(new Date(date.get("year"), date.get("month"), 1)).get("day");
+  }
+
+  const isLeapYear = (year: number) => {
+    return (year % 4 === 0 && year % 100 !== 0) || year % 400 === 0;
+  };
+
+  const activeVacation = (dayInfo: Date): any => {
```

Comment:
use memoization for this function

Reply:
i don't understand, what  i need to do here